pkg/reconciler/taskrun/local.go

```diff
@@ -0,0 +1,57 @@
+package taskrun
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	pipelinev1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type Local struct{}
+
+func (l Local) Allocate(r *ReconcileTaskRun, ctx context.Context, tr *pipelinev1.TaskRun, secretName string) (reconcile.Result, error) {
+	var err error
+	log := logr.FromContextOrDiscard(ctx)
+
+	log.Info("Task set to run locally in the cluster")
+	tr.Labels[AssignedHost] = "localhost"
+	controllerutil.AddFinalizer(tr, PipelineFinalizer)
+	if err = r.client.Update(ctx, tr); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	err = createUserTaskSecret(r, ctx, tr, secretName, map[string][]byte{
+		"host": []byte("localhost"),
+	})
```

Comment:
This is a convention that we are defining. Should we document this in the repo itself? Or will we just document it elsewhere?